src/frontend/markdown.rs

```diff
@@ -181,6 +187,189 @@ impl InlineRule for MathEquationScanner {
     }
 }
 
+// TOC SECTION
```

Comment:
This file is quite long, so you can make a folder markdown, move markdown.rs into markdown/mod.rs and then make a separate file for toc.